x-pack/plugin/core/template-resources/src/main/resources/elastic-connectors-sync-jobs.json

```diff
@@ -118,16 +122,16 @@
           "type": "date"
         },
         "deleted_document_count": {
-          "type": "integer"
+          "type": "long"
```

Comment:
Good to test if this is backward compatible change in a case of upgrade from ES with connector index already defined

Reply:
Thanks for flagging this. So this isn't backwards compatible, starting up after upgrading I get errors like:

```
Put mapping request for [.elastic-connectors-sync-jobs-v1] failed java.lang.IllegalArgumentException: mapper [total_document_count] cannot be changed from type [integer] to [long]
```

The reason I changed these are `long` is because that's what was being used in our tests (see [here](https://github.com/elastic/elasticsearch/blob/ff908d36118b0acba81f026b1f9025e0bac38deb/x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/connector/ConnectorTestUtils.java#L104) as an example). I didn't realise the previous mappings were `integer` until you pointed it out here.
Evidently the correct path forward is to fix the test data instead.

Reply:
My bad, the [connector protocol ](https://github.com/elastic/connectors/blob/main/docs/CONNECTOR_PROTOCOL.md) defined this as long so I implemented it like that ... I didn't realise the template mappings were different 

Nothing we can do about it now....

Reply:
> The connector class is more like a utility class, is not actually used to deserialize results from ES so nothing we should worry about

Do you mean it's probably okay to go forward with `integer`? (so long as we update the protocol)

Reply:
Created an issue: https://github.com/elastic/search-team/issues/9131
I'll cast the problem unit test types to `long` in the meantime.